### PR TITLE
Using display_user_id field to render username in web interface

### LINKED ISSUE #6498: Update web UI top bar to use new display_user_id field
## Summary

#6369 tracks the user who created the build (either manual trigger or rerun) and also tweaked the `api/v1/user` endpoint to contain a new `display_user_id` field. The new field should be used to show the user in the top bar, since it reflects the same value that Concourse tracks as the build creator.

We should be able to replace [this function](https://github.com/concourse/concourse/blob/37c4aebfb90fcbd1f7b7504846a9860f783b3a2d/web/elm/src/Login/Login.elm#L96-L102) with one that just uses the `display_user_id` field.

## Context

* Prior discussion: #6369

### PR DESCRIPTION
**Pull request by logyball**
Bug Fix, closes #6498 

## Changes proposed by this PR:
Updating [userDisplayName](https://github.com/concourse/concourse/blob/37c4aebfb90fcbd1f7b7504846a9860f783b3a2d/web/elm/src/Login/Login.elm#L96-L102) function in Login.elm to use the newly created user_display_id field from the /user API as per suggestion from @vito in #6498.

## Notes to reviewer:
In order to facilitate this dev, I had to touch a lot of the Concourse User object.  I'm still new to this, so hopefully I didn't do anything too silly.  I would like to write a test that actually validates this development in the `web/wats` directory, but since the functionality should remain unchanged, I would imagine the existing tests should simply not break.

## Contributor Checklist
- [x] Followed [Code of conduct], [Contributing Guide] & avoided [Anti-patterns]
- [x] [Signed] all commits
- [x] Added tests (Unit and/or Integration)
- [x] Updated [Documentation]

[Code of Conduct]: https://github.com/concourse/concourse/blob/master/CODE_OF_CONDUCT.md
[Contributing Guide]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md
[Anti-patterns]: https://github.com/concourse/concourse/wiki/Anti-Patterns
[Signed]: https://help.github.com/en/github/authenticating-to-github/signing-commits
[Documentation]: https://github.com/concourse/docs

## Reviewer Checklist
- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the
  [BOSH](https://github.com/concourse/concourse-bosh-release) and
  [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for
  the [integration
  tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env)
  (for example, if they are Garden configs that are not displayed in the
  `--help` text).

### CODE REVIEW

**Review by vito — CHANGES_REQUESTED** [review]
Sorry for the slow turnaround! Tried it out locally and couldn't get it to work, I think the precedence logic in `Login.elm` just needs to be tweaked.

File: web/elm/src/Login/Login.elm [hunk 1]
```diff
@@ -95,8 +95,8 @@ viewLoginState userState isUserMenuExpanded =
 
 userDisplayName : Concourse.User -> String
 userDisplayName user =
-    Maybe.withDefault user.id <|
+    Maybe.withDefault user.displayUserId <|
```
[the remark this answers is not in the document]
Comment (loganballard) [reply web/elm/src/Login/Login.elm:98]: Great idea.  Still learning the project and tech... thanks for the feedback! Added displayUserId as the first element in the list, returned id to default.

File: web/elm/tests/TopBarTests.elm [hunk 2]
```diff
@@ -856,6 +856,7 @@ all =
                                 , teams =
                                     Dict.fromList
                                         [ ( "t", [ "member" ] ) ]
+                                , displayUserId = "test"
```
[the remark this answers is not in the document]
Comment (loganballard) [reply web/elm/tests/TopBarTests.elm]: Made this a distinct value.  I believe the relevant test is actually using the Concourse User object defined further down.  See the latest commit

**Comment by loganballard** [comment]
Addressed @vito 's comments, requesting re-review.  No rush!

**Review by vito — APPROVED** [review]
lgtm - thanks!

**Comment by taylorsilva** [comment]
@logyball FYI we're enabling 2FA for the Concourse org. We can see you don't have 2FA enabled. You'll be kicked from the Concourse org when we flip the switch in a few minutes. If you enable 2FA then you can re-join the Concourse org.

**Pull request merged**